shoes-core/lib/shoes/common/initialization.rb

```diff
@@ -0,0 +1,40 @@
+class Shoes
+  module Common
+    module Initialization
+      def initialize(app, parent, *args)
+        blk    = args.pop if args.last.is_a?(Proc) || args.last.nil?
+        styles = args.last.is_a?(Hash) ? args.pop : {}
+
+        before_initialize(styles)
+
+        @app = app
+        @parent = parent
+
+        style_init(styles)
+        create_dimensions(*args)
+
+        @parent.add_child self
```

Comment:
Did I mention how I love to have this here and not add it EVERY DAMN TIME?

Thank you :heart: